### PR TITLE
added randomized lead time pretraining, calculation of rmse on train set

### PR DESCRIPTION
**Pull request by rohanshah13**

### CODE REVIEW

**Comment by jasonjewik** [comment]
Since I am traveling right now and you are merging into a non-main branch, I will defer to @tung-nd on this PR. If he says it is good, I'll approve. However since @prakhar6sharma is still listed as a code owner on the target branch, you also need his approval. I recommend removing him from the CODEOWNERS file to prevent this issue in the future.

**Review by tung-nd — APPROVED** [review]
looks good to me!

**Pull request merged**
